onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -240,7 +274,34 @@ def _onnx_infer_single_node(self, node):
             vi.CopyFrom(self.tmp_mp_.graph.output[i_o])
             self.known_vi_[vi.name] = vi
 
-    def _get_int_values(self, node):
+    def _onnx_infer_subgraph(self, node, subgraph):
+        if self.verbose_ > 2:
+            print('Inferencing subgraph of node {} with output({}...): {}'.format(node.name, node.output[0], node.op_type))
+        assert len(node.input) == len(subgraph.input)
+        subgraph_implicit_input = set()
+        for sn in subgraph.node:
+            subgraph_implicit_input.update([i for i in sn.input if i in self.known_vi_])
+        tmp_graph = helper.make_graph(list(subgraph.node),
```

Comment:
A value in the subgraph can shadow an outer scope value. Not sure that this takes that into account as I think it's checking outer scope values first for matches.